src/plugins/clearURLs/index.ts

```diff
@@ -51,42 +49,28 @@ export default definePlugin({
             : (str || "");
     },
 
-    createRules() {
-        // Can be extended upon once user configs are available
-        // Eg. (useDefaultRules: boolean, customRules: Array[string])
-        const rules = defaultRules;
-
-        this.universalRules = new Set();
-        this.rulesByHost = new Map();
-        this.hostRules = new Map();
-
-        for (const rule of rules) {
-            const splitRule = rule.split("@");
-            const paramRule = new RegExp(
-                "^" +
-                this.escapeRegExp(splitRule[0]).replace(/\\\*/, ".+?") +
-                "$"
-            );
+    async createRules() {
+        const url = "https://raw.githubusercontent.com/ClearURLs/Rules/master/data.min.json";
 
-            if (!splitRule[1]) {
-                this.universalRules.add(paramRule);
-                continue;
-            }
-            const hostRule = new RegExp(
-                "^(www\\.)?" +
-                this.escapeRegExp(splitRule[1])
-                    .replace(/\\\./, "\\.")
-                    .replace(/^\\\*\\\./, "(.+?\\.)?")
-                    .replace(/\\\*/, ".+?") +
-                "$"
-            );
-            const hostRuleIndex = hostRule.toString();
+        const response = await fetch(url);
+        const data = await response.json();
```

Comment:
```suggestion
        const res = await fetch(url).then(res => res.json());
```

also it would be good to type this json, you can easily do it with [this extension](https://marketplace.visualstudio.com/items?itemName=GregorBiswanger.json2ts)